src/app/api/organizations/[organizationId]/route.ts

```diff
@@ -6,28 +6,28 @@ import { zObjectId } from '@/types/dataModel/base';
 import { zUpdateOrganizationRequest } from '@/types/dataModel/organization';
 import { NextRequest, NextResponse } from 'next/server';
 import { mongo } from 'mongoose';
+import CMError, { CMErrorResponse, CMErrorType } from '@/utils/cmerror';
 
 // @route DELETE /api/organizations/[organizationId] - Soft deletes an organization
 export async function DELETE(
   _request: NextRequest,
   { params }: { params: { organizationId: string } }
 ) {
-  const validationResult = zObjectId.safeParse(params.organizationId);
-  if (!validationResult.success) {
-    return NextResponse.json(
-      { message: 'Invalid Organization Id' },
-      { status: 400 }
-    );
-  }
+  try {
+    const validationResult = zObjectId.safeParse(params.organizationId);
+    if (!validationResult.success) {
+      return new CMError(CMErrorType.BadValue, 'Organization Id').toNextResponse();
```

Comment:
@zaviermiller should it be an error if we try to soft delete an already soft-deleted organization?